### PR TITLE
add warn mode and update cosign

### PR DESCRIPTION
**Pull request by ltagliaferri**
publishing doc on warn vs fail mode for policies and cosign sign keyless updates from cosign 2.0

### CODE REVIEW

File: content/open-source/sigstore/cosign/an-introduction-to-cosign.md [hunk 1]
```diff
@@ -28,107 +28,115 @@ While code signing is not new technology, the growing prevalence of software in
...
 
-We’ll review Cosign commands here so you can understand how Cosign works in practice and we are using imagined demo containers. If you would like to follow along, please first [install Cosign](../how-to-install-cosign).
+Through offering short-lived credentials, keyless signing can support the recommended practice of operating your build environment like a production environment. This prevents the opportunity for long-lived keys to be stolen and used to sign malicious artifacts. Even if these short-lived keys used in keyless signing were stolen, they’d be useless!
 
-```sh
-cosign generate-key-pair 
-```
+While keyless signing can be used by individuals in the same manner as long-lived key pairs, it is also well suited for continuous integration and continuous deployment workloads. Keyless signing works by sending an OpenID Connect (OIDC) token to a certificate authority like Fulcio to be signed by a workload’s authenticated OIDC identity. This allows the developer to cryptographically demonstrate that the software artifact was built by the continuous integration pipeline of a given repository, for example.
```
Comment (jamonation) [inline content/open-source/sigstore/cosign/an-introduction-to-cosign.md:41]:
```suggestion
While keyless signing can be used by individuals in the same manner as long-lived key pairs, it is also well suited for continuous integration and continuous deployment workloads. Keyless signing works by sending an OpenID Connect (OIDC) token to a certificate authority like [Fulcio](/open-source/sigstore/fulcio) to be signed by a workload’s authenticated OIDC identity. This allows the developer to cryptographically demonstrate that the software artifact was built by the continuous integration pipeline of a given repository, for example. 
```

File: content/open-source/sigstore/cosign/an-introduction-to-cosign.md [hunk 2]
```diff
@@ -28,107 +28,115 @@ While code signing is not new technology, the growing prevalence of software in
...
-```
-Enter password for private key:
-Enter again:
-Private key written to cosign.key
-Public key written to cosign.pub
-```
+Cosign uses ephemeral keys and certificates, gets them signed automatically by the Fulcio root certificate authority, and stores these signatures in the Rekor transparency log, which automatically provides an attestation at the time of creation. 
 
-You can sign a container and store the signature in the registry with the cosign sign command. 
+You can manually create a keyless signature with the following command in cosign. In our example, we’ll use [Docker Hub](https://hub.docker.com/). If you would like to follow along, ensure you are logged into Docker Hub on your local machine and that you have a Docker repository with an image available. The following example assumes a username of `docker-username` and a repository name of `demo-container`.
```
Comment (jamonation) [inline content/open-source/sigstore/cosign/an-introduction-to-cosign.md:45]:
```suggestion
You can manually create a keyless signature with the following cosign command. In our example, we’ll use [Docker Hub](https://hub.docker.com/) to store the signature. If you would like to follow along, ensure you are logged into Docker Hub on your local machine and that you have a Docker repository with an image available. The following example assumes a username of `docker-username` and a repository name of `demo-container`. 
```

File: content/open-source/sigstore/cosign/an-introduction-to-cosign.md [hunk 3]
```diff
@@ -28,107 +28,115 @@ While code signing is not new technology, the growing prevalence of software in
...
+[{"critical":{"identity":{"docker-reference":"index.docker.io/docker-username/demo-container"},"image":{"docker-manifest-digest":"sha256:e..."},"type":"cosign container image signature"},"optional":{"1.3.6.1.4.1.57264.1.1":"https://accounts.google.com","Bundle":{"SignedEntryTimestamp":"...","Payload":{"body":"eyJhcGlWZX...X19","integratedTime":...,"logIndex":...,"logID":"..."}},"Issuer":"https://accounts.google.com","Subject":"username@gmail.com"}}]
+```
 
-Through offering short-lived credentials, keyless signing can support the recommended practice of operating your build environment like a production environment. This prevents the opportunity for long-lived keys to be stolen and used to sign malicious artifacts. Even if these short-lived keys used in keyless signing were stolen, they’d be useless!
+As part of the JSON output, you should get feedback on the issuer that you used and the email address associated with it. For example, if you used Google as the authenticator, you will have `"Issuer":"https://accounts.google.com","Subject":"username@gmail.com"}}]` as the last part of your output.
```
Comment (jamonation) [inline content/open-source/sigstore/cosign/an-introduction-to-cosign.md:98]:
```suggestion
As part of the JSON output, you should get feedback on the issuer that you used and the email address associated with it. For example, if you used Google as your OIDC provider, you will have `"Issuer":"https://accounts.google.com","Subject":"username@gmail.com"}}]` as the last part of your output. 
```

Comment (jamonation) [reply content/open-source/sigstore/cosign/an-introduction-to-cosign.md:98]: I think this helps reinforce the role of Google/MSFT/Github as the OIDC issuer in the whole signing flow?

File: content/open-source/sigstore/cosign/how-to-sign-a-container-with-cosign.md [hunk 4]
```diff
@@ -167,29 +121,33 @@ You should be able to now access your published container via your Docker Hub ac
 
 Now that the container is in a registry (in our example, it is in Docker Hub), you are ready to sign the container and push that signature to the registry.
 
-Make sure you are in the right directory for your Cosign key pair. 
+You will call your registry user name and container name with the following Cosign command. Note that we are signing the image in Docker Hub keylessly with Cosign.
```
Comment (jamonation) [inline content/open-source/sigstore/cosign/how-to-sign-a-container-with-cosign.md:124]:
```suggestion
You will call your registry user name and container name with the following cosign command. Note that we are signing the image in Docker Hub keylessly with cosign.
```

Comment (jamonation) [reply content/open-source/sigstore/cosign/how-to-sign-a-container-with-cosign.md:124]: I think there are more lowercase cosigns than Cosigns in other guides in this PR? I'm not sure which we prefer though.

**Review by jamonation — APPROVED** [review]
A couple minor nits and suggestions, thanks for updating all these!

File: content/open-source/sigstore/cosign/how-to-sign-a-container-with-cosign.md (hunk 4 above)
Comment (ltagliaferri) [reply content/open-source/sigstore/cosign/how-to-sign-a-container-with-cosign.md:124]: Yeah, the command should be lower-case but the project should be title case

**Pull request merged**